### DISCUSSION TITLE
Diagnostic messages disappear when save file

### DISCUSSION THREAD

**Discussion by kongjun18** [w0rp/ale#3721] [Q&A - Ask for help with problems]
I use both ALE and coc.nvim(ccls) under the instructions of README. When I save file, all diagnostic messages disappear. coc.nvim seem to lint code when leave insert mode and change in normal mode. When I type some text and then exit insert mode or modify in normal mode, all diagnostic messages appear.

How could I make ALE keep diagnostic messages?

My ALE configuration:
```
let g:ale_disable_lsp = 1
let g:ale_linters_explicit = 1 " Disable other linters
let g:ale_lint_on_text_changed = 'never'
let g:ale_lint_on_insert_leave = 1
nmap <silent> gN <Plug>(ale_previous_wrap)
nmap <silent> gn <Plug>(ale_next_wrap)
```

Screenshot：
![Diagnostic messages disappear when save file](https://user-images.githubusercontent.com/44012232/117778706-80af8580-b270-11eb-869b-1adf2b6c3dc6.gif)

**Comment by luisiacc** [comment]
I have this exact problem
My config is this one:
```viml
let g:ale_lint_on_text_changed = 'always'
let g:ale_fix_on_save = 1
let g:ale_fixers = {'javascript': ['prettier', 'eslint'], 'python': ['black', 'isort'], 'rust': ['rustfmt']}
let g:ale_disable_lsp = 1
let g:ale_set_highlights = 0
let g:ale_echo_msg_format = '[%linter%] (%code%) - %s'

let g:ale_linters_explicit = 1
let g:ale_linters = {'javascript': ['prettier', 'eslint'], 'python': ['flake8']}
let g:ale_lint_on_save = 1
let g:ale_lint_on_enter = 1

```

**Reply by luisiacc** [reply]
@kongjun18 did you get to fix this?

**Reply by luisiacc** [reply]
@w0rp is this known ?

**Comment by luisiacc** [comment]
Hi @kongjun18 , try setting those configs before loading ale plugin

**Reply by kongjun18** [reply]
Thank you for your generous help! It did not solve my problem.

**Reply by kongjun18** [reply]
Perhaps it is coc.nvim's problem.

**Reply by gialynguyen** [reply]
@kongjun18 have you solved it yet?

**Reply by kongjun18** [reply]
> @kongjun18 have you solved it yet?

I didn't solve it.

**Reply by hsanson** [reply]
Please try using the bare minimum configuration to discard problems caused by conflicts with other plugings. For example I always keep this custom vimrc for testing:

```
call plug#begin('$HOME/.config/nvim/bundle')
Plug '$HOME/Projects/vim/ale'
call plug#end()

let g:ale_virtualtext_cursor = 1
let g:ale_fixers = { 'ruby': ['rubocop'] }
let g:ale_linters = { 'ruby': ['rubocop'] }
```

I change the linters, fixers, and add/remove other ale options depending on what I am testing. Then I run vim/neovim using this config alone on a sample HelloWorld for the language I am testing:

```
nvim -u vimrc  hello.rb
```

If you are not using vim-plug, instead use runtimepath to add the ale plugin:

```
let g:ale_completion_enabled = 1

set runtimepath=$VIMRUNTIME
set runtimepath^=$HOME/Projects/vim/ale

let g:ale_virtualtext_cursor = 1
let g:ale_linters = {'python': ['pyright']}
```

Note that some ale configurations must be set before the plugin is loaded. Check the help for details on each config you are planning to use.

**Reply by hsanson** [reply]
Also providing the output of ALEInfo may allow other to help figuring out what may be wrong.

**Reply by gialynguyen** [reply]
I set `g:ale_lint_on_save = 0` and it works correctly. But when I set `g:ale_lint_on_save = 1`, after I save buffer, the signcolumn doesn't show errors when the linters detects errors
